src/utils/TriangleUtils.ts

```diff
@@ -61,56 +133,196 @@ let cleanIndex = function (ind, lengthArray) {
         res = ind % lengthArray;
     }
     return res;
-};
+},
 
 /**
- * A number, or a string containing a number.
- * @typedef {Object} VertexLike
- * @property {() => Vector3} getPos
- * @property {() => number} getThickness
+ * Updates the cached values of the triangle
+ * @param triangle The triangles who's internal values need to be updated
  */
+updateComputedAttributes(triangle: TriangleLike) {
+    let v0_p = triangle.v[0].getPos();
+    let v1_p = triangle.v[1].getPos();
+    let v2_p = triangle.v[2].getPos();
 
-/**
- * A number, or a string containing a number.
- * @typedef {Object} TriangleLike
- * @property {Array<VertexLike>} v
- * @property {Vector3} p0p1
- * @property {Vector3} p1p2
- * @property {Vector3} p2p0
- * @property {Vector3} unit_p0p1
- * @property {Vector3} unit_p1p2
- * @property {Vector3} unit_p2p0
- * @property {Vector3} unit_normal
- * @property {number} length_p0p1
- * @property {number} length_p1p2
- * @property {number} length_p2p0
- * @property {Vector3} unit_p0p1
- * @property {number} diffThick_p0p1
- * @property {number} diffThick_p1p2
- * @property {number} diffThick_p2p0
- * @property {Vector3} ortho_dir
- * @property {Vector3} point_min
- * @property {number} weight_min
- * @property {Vector3} main_dir
- * @property {Vector3} point_iso_zero
- * @property {Vector3} proj_dir
- * @property {boolean} equal_weights
- * @property {Vector3} half_dir_1
- * @property {Vector3} point_half
- * @property {Vector3} half_dir_2
- * @property {number} coord_max
- * @property {number} coord_middle
- * @property {number} unit_delta_weight
- * @property {Vector3} longest_dir_special
- * @property {number} max_seg_length = tmp.length();
- * @property {Vector3} unsigned_ortho_dir = triangle.ortho_dir.clone();
- */
+    triangle.p0p1 ? triangle.p0p1.subVectors(v1_p, v0_p) : null;
```

Comment:
Strange expression: you don't store the result anywhere. If you are interested in side effect only you should use an instruction instead.